### PR TITLE
[FEAT] Migrate iframe navigationFrameModule

### LINKED ISSUE #2583: Deprecation: #93944 - File Tree as iframe migrated to SVG-based tree
Deprecation: #93944 - File Tree as iframe migrated to SVG-based tree

https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/11.2/Deprecation-93944-FileTreeAsIframeMigratedToSVG-basedTree.html
.. include:: ../../Includes.txt

====================================================================
Deprecation: #93944 - File Tree as iframe migrated to SVG-based tree
====================================================================

See :issue:`93944`

Description
===========

When registered backend modules have the legacy-tree navigation frame
:html:`file_navframe` set via the module configuration
:php:`navigationFrameModule`, the modules are now using the new SVG-based Folder
tree view as component (non-iFrame).

Impact
======

Modules still registering the old :html:`file_navframe` option via
Module Configuration :php:`navigationFrameModule` will automatically be migrated
to the new Component, and a PHP :php:`E_USER_DEPRECATED` error will be triggered.

Affected Installations
======================

TYPO3 installations with custom extensions having backend modules
using the filelist navigation frame (folder-based tree).

Modules that use the implicit main module configuration and are
located directly within the "File" module are not affected.

Migration
=========

Change the affected code in your ext_tables.php:

:php:`'navigationFrameModule' => 'file_navframe'`

to

:php:`'navigationComponentId' => 'TYPO3/CMS/Backend/Tree/FileStorageTreeContainer'`

.. index:: Backend, NotScanned, ext:backend

### PR DESCRIPTION
**Pull request by helsner**
Resolves #2583

### CODE REVIEW

**Comment by helsner** [comment]
tbh dont know what to do with the failing phpstan as everything is fine on my local run

File: src/Rector/v11/v2/MigrateFrameModuleToSvgTreeRector.php [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [StaticCall::class];
+    }
+
+    public function refactor(Node $node): ?Node
```
Comment (sabbelasichon) [inline src/Rector/v11/v2/MigrateFrameModuleToSvgTreeRector.php:36]: Please add a docblock to specify the type in this case StaticCall

File: src/Rector/v11/v2/MigrateFrameModuleToSvgTreeRector.php [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    {
+        $fileInfo = $this->file->getSmartFileInfo();
+
+        if (! $this->filesFinder->isExtTables($fileInfo)) {
+            return null;
+        }
+
+        if (! $this->nodeTypeResolver->isMethodStaticCallOrClassMethodObjectType(
```
Comment (sabbelasichon) [inline src/Rector/v11/v2/MigrateFrameModuleToSvgTreeRector.php:44]: Please add a dedicated shouldSkip method for better readability with some early returns

File: src/Rector/v11/v2/MigrateFrameModuleToSvgTreeRector.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            &&
+            ! $this->nodeTypeResolver->isMethodStaticCallOrClassMethodObjectType(
+                $node,
+                new ObjectType('TYPO3\CMS\Core\Utility\ExtensionManagementUtility')
+            )
+        ) {
+            return null;
+        }
+
+        if (! (property_exists($node, 'name') && null !== $node->name && (property_exists(
```
Comment (sabbelasichon) [inline src/Rector/v11/v2/MigrateFrameModuleToSvgTreeRector.php:57]: If you have add the docblock there is no need for this code anymore, i guess.

File: src/Rector/v11/v2/MigrateFrameModuleToSvgTreeRector.php [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            return null;
+        }
+
+        if (! $this->isNames($node->name, ['addModule', 'registerModule'])) {
+            return null;
+        }
+
+        $hasAstBeenChanged = false;
+        if ($this->isName($node->name, 'addModule')) {
+            if (! (property_exists($node->args[4], 'value') && null !== $node->args[4]->value)) {
```
Comment (sabbelasichon) [inline src/Rector/v11/v2/MigrateFrameModuleToSvgTreeRector.php:70]: You could use the ValueResolver to get the value from the arguments so you could drop this additional checks here.

**Pull request merged**